src/main.py

```diff
@@ -219,7 +220,8 @@ def _generate_interaction_results(
             request.train_task_idx,
             request.termination_function,
             max_num_steps=CFG.max_num_steps_interaction_request,
-            monitor=monitor)
+            monitor=monitor,
+            exceptions_to_break_on={OptionExecutionFailure})
```

Comment:
why is this needed?

Reply:
+1

Reply:
Due to the following case in `get_interaction_requests()` in reinforcement_learning_approach.py:
```except (ApproachTimeout, ApproachFailure) as e:
                partial_refinements = e.info.get("partial_refinements")
                assert partial_refinements is not None
                _, plan = max(partial_refinements, key=lambda x: len(x[1]))
                _act_policy = utils.option_plan_to_policy(plan)
```
When the act policy is created from a list of options that is "incomplete"/ too short and does not reach the goal, there is a `OptionExecutionFailure("Option plan exhausted!")` that is thrown. 

Maybe we should fill this argument only when this particular approach is called?

Reply:
yes thanks! I believe we actually already merged this change for a similar reason in a different recent PR, so when you rebase off master, you should be all set here.

Reply:
Do to the following case in `get_interaction_requests()` in reinforcement_learning_approach.py: 
```
    partial_refinements = e.info.get("partial_refinements")
                assert partial_refinements is not None
                _, plan = max(partial_refinements, key=lambda x: len(x[1]))
                _act_policy = utils.option_plan_to_policy(plan)
```
Because the list of options that makes the policy is incomplete / too short, `OptionExecutionFailure("Option plan exhausted!")` is thrown in `option_plan_to_policy()`.

Is there a better way to populate this argument only when this particular approach is called?